### ISSUE TITLE
karpenter version is not compatible with K8s version 1.31

### ISSUE THREAD

**Issue by starlightromero** [aws/karpenter-provider-aws#7127]
### Description

**Observed Behavior**:

Karpenter Helm chart version `1.0.4` is deployed.

In the controller logs:
```
{"level":"ERROR","time":"2024-09-30T17:55:44.159Z","logger":"controller","caller":"amifamily/ami.go:91","message":"failed validating kubernetes version","commit":"0f8788c","controller":"nodeclass.status","controllerGroup":"karpenter.k8s.aws","controllerKind":"EC2NodeClass","EC2NodeClass":{"name":"bottlerocket-node"},"namespace":"","name":"bottlerocket-node","reconcileID":"REDACTED","error":"karpenter version is not compatible with K8s version 1.31"}
```

**Expected Behavior**:

The latest version of the Helm chart would support the latest Kubernetes version. Furthermore, there would be no error logs about version compatibility issues.

**Reproduction Steps** (Please include YAML):

```yaml
---
apiVersion: helm.toolkit.fluxcd.io/v2
kind: HelmRelease
metadata:
  name: karpenter
  namespace: karpenter-system
spec:
  chart:
    spec:
      chart: karpenter
      sourceRef:
        kind: HelmRepository
        name: karpenter
        namespace: karpenter-system
      version: 1.0.4
  driftDetection:
    mode: enabled
    # Ignore the replicas since we use VPA
    ignore:
    - paths: ["/spec/replicas"]
  install:
    crds: CreateReplace
  upgrade:
    crds: CreateReplace
  interval: 1m
  values:
    logLevel: debug
    replicas: 3
    dnsPolicy: Default
    serviceMonitor:
      enabled: true
    settings:
      clusterName: ${cluster_name}
    serviceAccount:
      annotations:
        eks.amazonaws.com/role-arn: arn:aws:iam::${aws_account_id}:role/${cluster_name}-karpenter-controller
    controller:
      resources:
        requests:
          cpu: 100m
          memory: 1Gi
        limits:
          cpu: 100m
          memory: 1Gi
```

**Versions**:
- Chart Version: 1.0.4
- Kubernetes Version (`kubectl version`): 1.31

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by jmdeal** [comment]
1.0.4 is not officially tested with Kubernetes 1.31 and only officially supports up to 1.30. We're in the process of updating our CI infrastructure to test with 1.31, once that is complete the release will support 1.31.

**Comment by sridevi501** [comment]
@starlightromero  1.0.5 released with karpenter 1.31 support.

**Comment by jmdeal** [comment]
Going to close this out now that the release has been made.
